### PR TITLE
refactor - clean dependency: core to devkit

### PR DESCRIPTION
**Pull request by elliotzh**
# Description

This PR mainly focus on clean the dependencies from core to devkit. Includes:
1. `core.Flow` & `core.AsyncFlow` have been simplified reimplemented without dependencies on devkit

Some connection related dependencies are left and marked for further refactoring.

Refactoring and renaming:

* [`src/promptflow/promptflow/_cli/_params.py`](diffhunk://#diff-a24d8278b691d05cb31b1f98b6456a2a0e8e7816bc79bbdf1b19fe45eaa8309dL19-R19): The import statement for `strip_quotation` has been updated to reflect its new location in the `promptflow._utils.utils` module.
* [`src/promptflow/promptflow/_cli/_pf/_init_entry_generators.py`](diffhunk://#diff-746b4bd84b2bf1ff736d6abe49639a025f3c7c3264f2001e331dafcffe2a5a0dL16-R16): The import statement for `FlowOperations` has been replaced with `is_executable_chat_flow` from `promptflow._utils.flow_utils`. Also, the method call to `FlowOperations._is_chat_flow` has been updated to `is_executable_chat_flow`. [[1]](diffhunk://#diff-746b4bd84b2bf1ff736d6abe49639a025f3c7c3264f2001e331dafcffe2a5a0dL16-R16) [[2]](diffhunk://#diff-746b4bd84b2bf1ff736d6abe49639a025f3c7c3264f2001e331dafcffe2a5a0dL234-R234)
* [`src/promptflow/promptflow/_core/connection_manager.py`](diffhunk://#diff-9e330c43ad9a3ea8a549e4015cf7972b9778afb4d01f3bdcc2d8f84ea37963acL12-R17): The import statement for `CustomStrongTypeConnectionConfigs` has been updated to reflect its new location in `promptflow._constants`.
* [`src/promptflow/promptflow/_proxy/_proxy_factory.py`](diffhunk://#diff-936478766b9de290e8791c802585d3932cfed416ae7b50daa76396dacf88cdf5R8-R15): The file was renamed from `src/promptflow/promptflow/batch/_executor_proxy_factory.py` and the `ExecutorProxyFactory` class was renamed to `ProxyFactory`. A new method `create_inspector_proxy` was added to the class. [[1]](diffhunk://#diff-936478766b9de290e8791c802585d3932cfed416ae7b50daa76396dacf88cdf5R8-R15) [[2]](diffhunk://#diff-936478766b9de290e8791c802585d3932cfed416ae7b50daa76396dacf88cdf5R54-R64)
* [`src/promptflow/promptflow/_sdk/_pf_client.py`](diffhunk://#diff-c100045cd304dbaa66dd61b7087f78b813ce01e8566d0b823b293bb35755eccfL18-R18): The import statement for `FlexFlow` has been updated to reflect its new location in `promptflow._sdk.entities._flow`.
* [`src/promptflow/promptflow/_sdk/_submitter/run_submitter.py`](diffhunk://#diff-e3499cf713c66612d91723759de1c5076c22689b2ce2c89678f82de624b57c2cL12-R16): The import statement for `parse_variant` has been updated to reflect its new location in `promptflow._utils.flow_utils`. The import statement for `FlexFlow` has been updated to reflect its new location in `promptflow._sdk.entities._flow`. The method call to `ExecutorProxyFactory().get_executor_proxy_cls(flow.language).dump_metadata` has been updated to `ProxyFactory().get_executor_proxy_cls(flow.language).dump_metadata`. [[1]](diffhunk://#diff-e3499cf713c66612d91723759de1c5076c22689b2ce2c89678f82de624b57c2cL12-R16) [[2]](diffhunk://#diff-e3499cf713c66612d91723759de1c5076c22689b2ce2c89678f82de624b57c2cL26-R26) [[3]](diffhunk://#diff-e3499cf713c66612d91723759de1c5076c22689b2ce2c89678f82de624b57c2cL110-R116)
* [`src/promptflow/promptflow/_sdk/_submitter/test_submitter.py`](diffhunk://#diff-d9dd2c09a149b11eb54626edf065287eeb7b1a28e39719b8dd95377770f64138R14-R20): The import statement for `dump_flow_result` has been updated to reflect its new location in `promptflow._utils.flow_utils`. The import statement for `FlexFlow` has been updated to reflect its new location in `promptflow._sdk.entities._flow`. The method call to `ExecutorProxyFactory().get_executor_proxy_cls(self.flow.language).dump_metadata` has been updated to `ProxyFactory().get_executor_proxy_cls(self.flow.language).dump_metadata`. [[1]](diffhunk://#diff-d9dd2c09a149b11eb54626edf065287eeb7b1a28e39719b8dd95377770f64138R14-R20) [[2]](diffhunk://#diff-d9dd2c09a149b11eb54626edf065287eeb7b1a28e39719b8dd95377770f64138R31-R35) [[3]](diffhunk://#diff-d9dd2c09a149b11eb54626edf065287eeb7b1a28e39719b8dd95377770f64138L248-R225)
* [`src/promptflow/promptflow/_sdk/_submitter/utils.py`](diffhunk://#diff-38c78274ff4a19381a2129bfd86174d225e0d8537b48c80aa8f6feaca25083abR28): The import statement for `ProxyFactory` from `promptflow._proxy` has been added. The method `resolve_connections` has been updated to use `ProxyFactory` and `resolve_connection_names`. The method `resolve_used_connections` has been removed. [[1]](diffhunk://#diff-38c78274ff4a19381a2129bfd86174d225e0d8537b48c80aa8f6feaca25083abR28) [[2]](diffhunk://#diff-38c78274ff4a19381a2129bfd86174d225e0d8537b48c80aa8f6feaca25083abL255-R281)

Addition of new files:

* [`src/promptflow/promptflow/_proxy/__init__.py`](diffhunk://#diff-db890b29f2c05ea85ec282550510ac0b0e4761b6be77af142b2e659d4e5a00b0R1-R8): This new file was added to import `ProxyFactory` and `AbstractInspectorProxy` from the `_proxy` module.
* [`src/promptflow/promptflow/_proxy/_base_inspector_proxy.py`](diffhunk://#diff-69df25f8ddfbeed622ae5cf13f97f439110586d51262ed1201ce5f3f30f18231R1-R27): This new file was added to define the `AbstractInspectorProxy` class.
* [`src/promptflow/promptflow/_proxy/_csharp_inspector_proxy.py`](diffhunk://#diff-3a3b6fcbb15ab97d02280e29888e85dc5c50ca6a78f930442b9fd83a5e219a31R1-R41): This new file was added to define the `CSharpInspectorProxy` class.
* [`src/promptflow/promptflow/_proxy/_python_inspector_proxy.py`](diffhunk://#diff-a362f804bcd889c4254da0b3249def515e14306ff5dfbf951dff3d8c12b4e7a2R1-R17): This new file was added to define the `PythonInspectorProxy` class.

# All Promptflow Contribution checklist:
- [x] **The pull request does not introduce [breaking changes].**
- [x] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](../CONTRIBUTING.md).**
- [x] **Create an issue and link to the pull request to get dedicated review from promptflow team. Learn more: [suggested workflow](../CONTRIBUTING.md#suggested-workflow).**

## General Guidelines and Best Practices
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### Testing Guidelines
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: src/promptflow/promptflow/_sdk/entities/_flow/async_dag.py [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+from promptflow._constants import DEFAULT_ENCODING, FlowLanguage
+from promptflow._utils.docs import AsyncFlowDoc
+from promptflow._utils.yaml_utils import load_yaml_string
+from promptflow.exceptions import UserErrorException
+
+from .dag import Flow
+
+
+class AsyncFlow(Flow):
+    __doc__ = AsyncFlowDoc.__doc__
```
Comment (crazygao) [inline src/promptflow/promptflow/_sdk/entities/_flow/async_dag.py:13]: Code editors cannot resolve this __doc__.

File: src/promptflow/tests/sdk_cli_test/e2etests/test_flow_as_func.py [hunk 2]
```diff
@@ -62,7 +62,7 @@ async def test_flow_as_a_func_asynckw(self, async_call_folder):
     async def test_flow_as_a_func_real_async(self):
         from promptflow.core._flow import AsyncFlow
 
-        original_async_func = AsyncFlow.invoke_async
+        original_async_func = AsyncFlow.invoke
```
Comment (crazygao) [inline src/promptflow/tests/sdk_cli_test/e2etests/test_flow_as_func.py:65]: Can this work properly?

File: src/promptflow/promptflow/_sdk/entities/_flow/async_dag.py (hunk 1 above)
Comment (elliotzh) [reply src/promptflow/promptflow/_sdk/entities/_flow/async_dag.py:13]: then let's just copy the doc string for now

File: src/promptflow/tests/sdk_cli_test/e2etests/test_flow_as_func.py (hunk 2 above)
Comment (elliotzh) [reply src/promptflow/tests/sdk_cli_test/e2etests/test_flow_as_func.py:65]: looks good with current tests

**Review by thy09 — APPROVED** [review]

File: src/promptflow/promptflow/core/_flow.py [hunk 3]
```diff
@@ -196,35 +47,40 @@ def load(
...
-        if isinstance(other, Flow):
-            return self._content_hash == other._content_hash and self.context == other.context
-        return False
+class Flow(FlowBase):
+    """A Flow in the context of PromptFlow is a sequence of steps that define a task.
+    Each step in the flow could be a prompt that is sent to a language model, or simply a function task,
+    and the output of one step can be used as the input to the next.
+    Flows can be used to build complex applications with language models.
+
+    Simple Example:
```
Comment (wangchao1230) [inline src/promptflow/promptflow/core/_flow.py:73]: Simple Example: ->  Example:

**Review by wangchao1230 — APPROVED** [review]

File: src/promptflow/promptflow/core/_flow.py [hunk 4]
```diff
@@ -196,35 +47,40 @@ def load(
...
             An exception is raised if the file does not exist.
         :type source: Union[PathLike, str]
-        :param raise_error: Argument for non-dag flow raise validation error on unknown fields.
-        :type raise_error: bool
         :return: A Flow object
         :rtype: Flow
         """
-        _, flow_path = cls._load_prepare(source)
+        flow_dir, flow_filename = resolve_flow_path(source)
+        flow_path = flow_dir / flow_filename
```
[the remark this answers is not in the document]
Comment (elliotzh) [reply src/promptflow/promptflow/core/_flow.py:54]: core shouldn't be aware of csharp but we can check if language is python

**Pull request merged**